cyphal_can.lua

```diff
@@ -0,0 +1,241 @@
+cyphal_getinfo_software_image_crc = ProtoField.uint64("cyphal.getinfo.response.crc", "CRC-64-WE", base.HEX)
+cyphal_getinfo_certificate_of_authority = ProtoField.bytes("cyphal.getinfo.response.certificate_of_authority", "Certificate of Authority")
+
+cyphal_can.fields = {
+    cyphal_can_id,
+    cyphal_can_priority,
+    cyphal_can_len,
+    cyphal_can_service_not_message,
+    cypahl_can_request_not_response,
+    cyphal_can_anonymous,
+    cyphal_can_subject_id,
+    cyphal_can_service_id,
+    cyphal_can_destination_node_id,
+    cyphal_can_source_node_id,
+    cyphal_can_tail_byte,
+    cyphal_can_start_of_transfer,
+    cyphal_can_end_of_transfer,
+    cyphal_can_toggle,
+    cyphal_can_transfer_id,
+    cyphal_can_crc,
+    -- heartbeat
+    cyphal_heartbeat_uptime,
+    cyphal_heartbeat_health,
+    cyphal_heartbeat_mode,
+    cyphal_heartbeat_vssc,
+    -- GetInfo
+    cyphal_getinfo_protocol_version_major,
+    cyphal_getinfo_protocol_version_minor,
+    cyphal_getinfo_hardware_version_major,
+    cyphal_getinfo_hardware_version_minor,
+    cyphal_getinfo_software_version_major,
+    cyphal_getinfo_software_version_minor,
+    cyphal_getinfo_software_vcs_revision_id,
+    cyphal_getinfo_unique_id,
+    cyphal_getinfo_name,
+    cyphal_getinfo_software_image_crc,
+    cyphal_getinfo_certificate_of_authority,
+    -- Add more fields
+
+}
+
+-- Function to dissect the CYPHAL/CAN
+function cyphal_can.dissector(buffer, pinfo, tree)
+
+    -- Extract the alrady found CAN ID
+    local can_id_fieldinfo = can_id_field()
+    local can_id = can_id_fieldinfo()
+    -- SocketCAN converts the DLC to a Length for you
+    local can_len_fieldinfo = can_len_field()
+    local can_len = can_len_fieldinfo()
+    -- we'll dissect the rest too potentially
+    local payload_len = buffer:len() - 1
+    local payload = buffer(0, payload_len)
+    local tail_buffer = buffer(payload_len, 1)
+
+    -- Create a subtree for the custom protocol
+    local header_tree = tree:add(cyphal_can, "Cyphal/CAN Header") -- no buffer() since it's over the existing CAN ID
+    local payload_tree = tree:add(cyphal_can, payload, "Cyphal/CAN Payload")
+    local footer_tree = tree:add(cyphal_can, tail_buffer, "Cyphal/CAN Footer")
+
+    -- common header parts
+    header_tree:add(cyphal_can_len, payload_len)
+    header_tree:add(cyphal_can_priority, can_id)
+    header_tree:add(cyphal_can_service_not_message, can_id)
+    -- Tail Byte Decode
+    local tail_byte = tail_buffer:uint()
+    footer_tree:add(cyphal_can_start_of_transfer, tail_buffer, tail_byte)
+    footer_tree:add(cyphal_can_end_of_transfer, tail_buffer, tail_byte)
+    footer_tree:add(cyphal_can_toggle, tail_buffer, tail_byte)
+    footer_tree:add(cyphal_can_transfer_id, tail_buffer, tail_byte)
+    local sot = bit.rshift(bit.band(tail_byte, 0x80), 7)
+    local eot = bit.rshift(bit.band(tail_byte, 0x40), 6)
+    if sot == 0 and eot == 1 then
+      payload_tree:add(cyphal_can_crc, payload(payload_len - 2, 2))
```

Comment:
Is this correct, considering that the CRC is encoded in the big-endian format?

Reply:
Lua Wireshark filters are by default BE unless you specifically add as LE.